app/models/measurement.rb

```diff
@@ -73,9 +70,23 @@ class Measurement < ActiveRecord::Base
   end)
 
   scope(:with_time, lambda do |data|
-    day_range(data[:day_from], data[:day_to]).
-      time_range(data[:time_from], data[:time_to]).
-      year_range(Date.new(data[:year_from].to_i),Date.new(data[:year_to].to_i + 1) - 1)
+    time_from = Time.strptime(data[:time_from].to_s, '%s')
+    time_to = Time.strptime(data[:time_to].to_s, '%s')
+
+    where("DATE(time) >= ?", time_from.beginning_of_day)
+      .where("DATE(time) <= ?",  time_to.end_of_day)
+      .minutes_range(Session.minutes_of_day(time_from), Session.minutes_of_day(time_to))
+  end)
+
+  scope(:minutes_range, lambda do |minutes_from, minutes_to|
```

Comment:
Do we need to keep this query separated from `with_time` or could they be one scope?